src/Akeneo/Connectivity/Connection/back/Application/ErrorManagement/Service/CollectApiError.php

```diff
@@ -51,14 +50,12 @@ public function __construct(
     }
 
     public function collectFromProductDomainError(
-        \Throwable $error,
-        ?ProductInterface $product
+       \Throwable $error,
+       Context $context
```

Comment:
```suggestion
       array $context
```
I will rather have an array with the right phpdoc `array{product: object|null}` then use the `Context` object that is an internal to the use of the serializer

Reply:
You're right, but I think we can let it like that for the moment: we will rework this part during decoupling (https://github.com/akeneo/pim-community-dev/blob/78efc2e1e1584b2283be7a88d8561f2eb036ab73/src/Akeneo/Connectivity/Connection/back/tests/.php_cd.php#L34)